notes/lecture11.tex

```diff
@@ -0,0 +1,130 @@
+}
+
+In this lecture we take a look of machine learning, and empirical risk minimization in particular. We define the distribution of our data as $D$ over $X\times Y$, $X\subseteq \R^{n}, \,Y = \{ -1,1 \}$
+\begin{itemize}
+\item "Model" is specified by a set of parameters $w \in \domain \in \mathbb{R}^{n}$
+\item "Loss function" $L: \domain \times (X\times Y) \rightarrow \R$, note that $l(w,z)$ gives the loss of model $w$ on instance $z$
+\item Population objective (\textit{Risk}): $R(w)=\underset{z\in D}{\mathbb{E}}[l(w,z)]$
+\item Goal: Find $w$ that minimizes $R(w)$
+\end{itemize}
+
+One way to accomplish this is to use stochastic optimization:
+$$w_{t+1} = w_{t} - \eta \nabla \ell(w_t,z_t) \quad z\in D $$
+\subsection{Empirical Risk}
+Suppose $S\in (X \times Y)^{m}, \,\, S=((x_1,y_1),......,(x_m,y_m))$, and $z_i$ represents the instance $(x_i,y_i), i\in \{1, ...,m \}$.The empirical risk is define as:
+$$R_{s}(w) = \frac{1}{m}\sum_{i=1}^{m}\ell(w,z_i)$$
+Our goal is to minimize this empirical risk. However, what we really want is : $R_{s}(w) = R(w)$.
+\begin{itemize}
+\item $R(w)$ captures loss on unseen example 
+\item $R_s(w)$ captures loss on seen example
+\end{itemize}
+\begin{definition}[Generalization error]
+$$\mathcal{E}_{\text{gen}}(w) = R(w) - R_{s}(w)$$
+Then: $R(w) = R_{s}(w) + \mathcal{E}_{\text{gen}}(w)$, where optimization can handle the part $R_{s}(w)$ pretty well.
+\end{definition}
+\textbf{How do we bound $\mathcal{E}_{\text{gen}}(w)$ ?}
+\begin{itemize}
+\item Principle: generalization error = stability
+\item Stability: How much does your model change if you change one training point
+\end{itemize}
+
+Choose Sample:
+
+$$S = (z_1, ... , z_m) \quad \quad  S^{\prime} = (z^{\prime}_1, ... ,z^{\prime}_{m})$$ 
+$S$ and $S^{\prime}$ can be completely different, we denote $S^{(i)}$ as:
+$$S^{(i)} = (z_1, \,... ,\, z_{i-1}, \,\underline{\mathbf{z_{i}^{\prime}}},\, z_{i+1}, \,... \,,z_{m})$$
+\begin{definition}[Average stability] The average stability of an algorithm $A: (X,Y)^{m} \rightarrow \domain$:
+$$\Delta(A) = \mathbb{E}[\frac{1}{m}\sum_{i=1}^{m}[\ell(A(s),z_i^{\prime}) - \ell(A(s^{(i)}),z_{i}^{\prime})]]$$
+\end{definition}
+This can be interpreted as performance on something unseen versus something seen.
+\begin{theorem}
+$$\mathbb{E}[\mathcal{E}_{\text(gen)}(A)] = \Delta(A)$$
+\end{theorem}
+\begin{proof}
+\begin{align*}
+\mathbb{E}[\mathcal{E}_{\text(gen)}(A)]  &= R(A(s)) - R_{s}(A(s)) \\
+\mathbb{E} [R_{s}(A(s))] &= \mathbb{E}[\frac{1}{m}\sum_{i=1}^{m}\ell(A(s),z_i)] \\
+\mathbb{E} [R(A(s))]&= \mathbb{E}[\frac{1}{m}\sum_{i=1}^{m}\ell(A(s),z_i^{\prime})]
+\end{align*}
+Hence, since $\mathbb{E}[\ell(A(s),z_i)] = \mathbb{E}\ell(A(s^{(i)}),z_i^{\prime})$
+$\mathbb{E}[R - R_s] = \Delta(A)$
+\end{proof}
+
+
+
+
+
+\subsection{Uniform Stability}
+We would now show how stability is related to generalization. For doing so we need to define the concept of \textit{uniform stability}.
```

Comment:
We already related stability to generalization earlier. Perhaps explain that uniform stability is convenient to work with and still leads to reasonable upper bounds.